### PR TITLE
Fix typo in the name of CapacityBaseRandomPolicy

### PR DESCRIPTION
**Pull request by dbw9580**
Fix typo in the name of CapacityBaseRandomPolicy.

Deprecate `CapacityBaseRandomPolicy` and add `CapacityBasedRandomPolicy`.

### CODE REVIEW

**Comment by dbw9580** [comment]
This needs rebasing after #16187 and #16237 are merged.

File: core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java [hunk 1]
```diff
@@ -11,119 +11,33 @@
...
-    }
-    List<WorkerNetAddress> cacheWorkers = mBlockLocationCache.getIfPresent(blockId);
-    if (cacheWorkers == null) {
-      cacheWorkers = new CopyOnWriteArrayList<>();
-      // guava cache is thread-safe
-      mBlockLocationCache.put(blockId, cacheWorkers);
-    }
-    cacheWorkers.add(targetWorker);
+    super(conf);
+    LOG.warn("This class name contains a typo. Use CapacityBasedRandomPolicy instead.");
```
Comment (jiacheliu3) [inline core/client/fs/src/main/java/alluxio/client/block/policy/CapacityBaseRandomPolicy.java:41]:
```suggestion
    LOG.warn("This class name contains a typo and will be deprecated in 3.0. Use {} instead.", CapacityBasedRandomPolicy.getClass());
```

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment

**Comment by jiacheliu3** [comment]
@dbw9580 rebase and merge?

**Pull request closed without merging**